snapcraft/internal/project_loader/grammar_processing/_part_grammar_processor.py

```diff
@@ -58,6 +58,12 @@ def __init__(self, *, plugin, properties, project_options, repo):
         self._project_options = project_options
         self._repo = repo
 
+        for key, value in properties.items():
+            if key in ['source']:
```

Comment:
is this just to support future expansion? I wouldn't complicate it now unless we know there is a very close next candidate to add.

Reply:
The next candidate is `source-checksum` which I think is paramount to have for different sources depending on the architecture.